### PR TITLE
Add test case for new ReferenceError after removal of per-parameter position var scope

### PR DESCRIPTION
**Pull request by spectranaut**
This test case is in response to the changes in https://github.com/tc39/ecma262/pull/1046 and discussion https://github.com/tc39/test262/pull/2445#discussion_r358789570

### CODE REVIEW

**Comment by spectranaut** [comment]
@syg this is how we generate tests for all function forms! Can you take a look at the case file and make sure the info quote and desc are reasonable?

File: src/function-forms/eval-var-scope-ref-err.case [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  1. Return the result of performing IteratorBindingInitialization for BindingElement with arguments iteratorRecord and environment.
+
+
+features: [default-parameters]
+---*/
+
+//- params
+a = eval("var a = 42")
+//- error
+ReferenceError
```
Comment (syg) [inline src/function-forms/eval-var-scope-ref-err.case]: Redeclarations are `SyntaxError`s not `ReferenceError`s.

**Comment by patrick-soquet** [comment]
My understanding was that the specification change introduced a new ReferenceError in sloppy mode. But the committed test cases check for SyntaxError in both sloppy and strict modes. Did I miss something? Thank you

**Comment by ljharb** [comment]
See #2460, i think

**Comment by syg** [comment]
> My understanding was that the specification change introduced a new ReferenceError in sloppy mode. But the committed test cases check for SyntaxError in both sloppy and strict modes. Did I miss something? Thank you

They are indeed sloppy only. I wasn't aware of the noStrict flag and didn't know all tests ran both strict/sloppy.

**Comment by patrick-soquet** [comment]
Since class code is strict mode code, it looks like the following test cases are incorrect:
```
language/expressions/class/async-gen-method/eval-var-scope-syntax-err.js
language/expressions/class/async-gen-method-static/eval-var-scope-syntax-err.js
language/expressions/class/gen-method/eval-var-scope-syntax-err.js
language/expressions/class/gen-method-static/eval-var-scope-syntax-err.js
language/expressions/class/method/eval-var-scope-syntax-err.js
language/expressions/class/method-static/eval-var-scope-syntax-err.js
language/statements/class/async-gen-method/eval-var-scope-syntax-err.js
language/statements/class/async-gen-method-static/eval-var-scope-syntax-err.js
language/statements/class/gen-method/eval-var-scope-syntax-err.js
language/statements/class/gen-method-static/eval-var-scope-syntax-err.js
language/statements/class/method/eval-var-scope-syntax-err.js
language/statements/class/method-static/eval-var-scope-syntax-err.js
```
I modified XS to conform to the specification change and to throw a SyntaxError. The new version will be published soon. Thank you.

**Comment by syg** [comment]
@patrick-soquet good catch! @spectranaut is there a way to have the `noStrict` flag passed only to a subset of generated tests? I don't know how to work the python test generator.

**Pull request merged**